### ISSUE TITLE
Add Equity Vol product

### ISSUE THREAD

**Issue by LaGuiche55** [domokane/FinancePy#117]
Hi

Is it possible to add the following popular equity vol product?
equity volatility swap
equity corridor variance swap
equity forward volatility agreement

Regards
JP

**Comment by rkcah** [comment]
Cool. Will take a look at this. I am not very familiar with these products so will have to read a bit about these. Is there any good material that you are aware on these?

**Comment by domokane** [comment]
Is there a specific asset class or product that you feel close to being a specialist in ? That would probably be best.

**Comment by domokane** [comment]
I have implemented variance swaps. Here is a description of volatility swaps by Derman. 

https://www.researchgate.net/publication/246869706_More_Than_You_Ever_Wanted_to_Know_About_Volatility_Swaps

**Comment by rkcah** [comment]
I usually work on rates products but would definitely prefer working on this since it's a bit out of my comfort zone. 

I read about Equity Volatility Swaps, seems like these are pretty standard forwards on volatility. Will start working on these.

**Comment by domokane** [comment]
Sounds good!

**Comment by LaGuiche55** [comment]
The first natural product is the variance swap for which you can find a similar paper here: https://www.google.com/url?sa=t&source=web&rct=j&url=https://www.sk3w.co/documents/volatility_trading.pdf&ved=2ahUKEwjt67eqx9X2AhXTa8AKHQh2BFoQFnoECCgQAQ&usg=AOvVaw0KihO920Ycok9REeVjtRaa

For the variance swap and the volswap, there is usually a cap where the payoff does not pay more than 2.5 times the strike. There is as well an uncapped version.

The variance swap is the first "natural" product because it can be replicated with vanillas making some assumptions. The traded price will be this vanilla replication + basis. The volatility swap will be derived from the variance swap with a convexity adjustment coming from the vol of vol.

Here you can find some intuition about the subtle difference between different volatility instruments
https://quant.stackexchange.com/questions/29830/is-the-vix-more-similar-to-a-volatility-swap-or-a-variance-swap

**Comment by LaGuiche55** [comment]
For corridor variance or conditionnal variance swap, I think there is broadly a confusion on which one is which if you talk to different traders. A source of information is
https://fincad.com/resources/resource-library/article/next-generation-variance-derivatives

Broadly it is a variance swap that accrues realised variance on day T only if the spot EOD T is within a spot range 80%-120% of initial spot.
The difference between corridor and conditional is if the nb of days denominator is rescaled for the number of days in the range or not.
The spot range can be 70-130, 80-120, 70-plus inf, 0-130 or any custom range.
The spot condition can be spot T in the range or spot T & T-1 in the range.
A more exotic version contains a Knock Out like corridor T & T-1, 70-110, KO 120
https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3383005

More exotic versions would be cross corridor long Nikkei short SPX where the range is on SPX.
https://www.google.com/url?sa=t&source=web&rct=j&url=https://www.mathfinance.com/wp-content/uploads/2018/04/Day1_07_Liang_CVSS.pdf&ved=2ahUKEwiv2oK9zNX2AhXUX8AKHTOdCokQFnoECDIQAQ&usg=AOvVaw2gs021BC-xyGgTb63PGIvE

**Comment by domokane** [comment]
Yes. If you can do a vol swap first maybe. I have already implemented the variance swap which can be replicated using dynamic hedging plus a log contract replication using calls and puts.
